components/cross_table_league/commons/cross_table_league_base.lua

```diff
@@ -0,0 +1,532 @@
+				name = opponentInput,
+				flag = args[prefix .. 'flag'],
+				link = args[prefix .. 'link'],
+			}}}
+		end
+		opponent = Opponent.resolve(Opponent.readOpponentArgs(opponent), self.config.date, {syncPlayer = true})
+		opponent.name = opponent.name or Opponent.toName(opponent)
+		table.insert(entries, {
+			opponent = opponent,
+			aliases = self:_processAliases(opponent.type, args[prefix .. 'alias'], opponentInput)
+		})
+	end
+
+	return entries
+end
+
+---@param opponentType string
+---@param aliasInput string?
+---@param opponentInput string
+---@return string[]
+function CrossTableLeague:_processAliases(opponentType, aliasInput, opponentInput)
+	if not aliasInput then return {} end
+	local aliases = mw.text.split(aliasInput, ',')
+
+	if opponentType ~= Opponent.team then
+		return aliases
+	end
+
+	aliases = Array.map(aliases, function(alias) return Team.page(nil, alias, self.config.date) end)
+
+	local allAliases = {}
+	Array.forEach(aliases, function(alias)
+		Array.extendWith(allAliases, {alias}, Team.queryHistoricalNames(alias:gsub('_', ' '):lower()))
+	end)
+	--for team input the `opponentInput` is supposed to be a team template, query historical names for it as aliases too
+	Array.extendWith(allAliases, Team.queryHistoricalNames(opponentInput:gsub('_', ' '):lower()))
+
+	return allAliases
+end
+
+---@return self
+function CrossTableLeague:query()
+	local config = self.config
+
+	local conditions = {TournamentStructure.getMatch2Filter(config.matchGroupSpecs)}
+
+	local toOpponentCondition = function(name)
+		return '[[opponent::' .. name .. ']]'
+	end
+
+	local opponentConditions = {}
+	Array.forEach(self.opponents, function(entries)
+		Array.forEach(entries, function(entry)
+			table.insert(opponentConditions, toOpponentCondition(entry.opponent.name))
+			Array.extend(opponentConditions, Array.map(entry.aliases, toOpponentCondition))
+		end)
+	end)
+	if Table.isNotEmpty(opponentConditions) then
+		table.insert(conditions, '(' .. table.concat(opponentConditions, ' OR ') .. ')')
+	end
+
+	if config.startDate then
+		table.insert(conditions, '([[date::>' .. config.startDate .. ']] OR [[date::' .. config.startDate .. ']])')
+	end
+
+	if config.endDate then
+		table.insert(conditions, '([[date::<' .. config.endDate .. ']] OR [[date::' .. config.endDate .. ']])')
+	end
+
+	if not config.dateLess then
+		table.insert(conditions, '([[date::!1970-01-01 00:00:00]] OR [[finished::1]])')
+	end
+
+	self.matches = self:_filterAndSortMatches(mw.ext.LiquipediaDB.lpdb('match2', {
+		limit = 1000,
+		offset = 0,
+		order = config.queryOrder,
+		conditions = table.concat(conditions, ' AND '),
+	}))
+
+	return self
+end
+
+---@param match table
+---@return table
+function CrossTableLeague._flipMatch(match)
+	local copy = Table.deepCopy(match)
+	--switch opponents; scores are inside opponents and switch too
+	copy.match2opponents[1] = match.match2opponents[2]
+	copy.match2opponents[2] = match.match2opponents[1]
```

Comment:
Does a few other fields also need to be swapped? like `winner`?

Reply:
no, opponents hold the scores etc hence they are swapped by this too
as for `match.winner`: the only usage for that is checking if it is `0` which is indep. of opponent order

Reply:
I would prefer to either invert the winner (keep it correct), or remove nil it